src/WebJobs.Script.WebHost/WebHost/IScriptWebHostEnvironment.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public interface IScriptWebHostEnvironment
+    {
+        /// <summary>
+        /// Gets a value indicating whether requests should be delayed.
+        /// </summary>
+        bool DelayRequestsEnabled { get; }
+
+        /// <summary>
+        /// Gets a <see cref="Task"> that will complete when the requests no longer need to be delayed.
+        /// </summary>
+        Task DelayCompletionTask { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the current environment is in standby mode.
+        /// </summary>
+        bool InStandbyMode { get; }
+
+        /// <summary>
+        /// Flags that requests under this environment should be delayed.
+        /// </summary>
+        void DelayRequests();
+
+        /// <summary>
+        /// Flags that requests under this environment should be resumed.
+        /// </summary>
+        void ResumeRequests();
+
+        /// <summary>
+        /// Flags the current environment as ready and specialized.
+        /// This sets <see cref="EnvironmentSettingNames.AzureWebsitePlaceholderMode"/> to "0"
```

Comment:
nit: no need to list the implementation details here since this is an interface method

Reply:
I didn't include such a method previously since these flags get set at different times by different components. You added the interface method to reduce code duplication? I believe it was the Linux codepaths that had to set these - on Windows it happens externally